File: 3. Herencia, extension e implementacion/En C#/VasoDeChupitoDeCristal.cs
using System;

namespace model 
{
    class VasoDeChupitoDeCristal : AVaso, ICrystalGlass {
        public VasoDeChupitoDeCristal(double capacity, double weight) : base(ICrystalGlass.getMaterial(), capacity, weight, "Vaso de chupito")
        {
        }
    }

}

File: 3. Herencia, extension e implementacion/En Java/hei/src/model/VasoDeChupitoDeCristal.java
package model;

public class VasoDeChupitoDeCristal extends AVaso implements ICrystalGlass {

    public VasoDeChupitoDeCristal(double capacity, double weight) {
        super(ICrystalGlass.material(), capacity, weight, "Vaso de chupito");
    }
    
}
